### PR TITLE
Deprecate Schedule instructions for Timed instructions

### PR DESCRIPTION
**Pull request by taalexander**
### Summary
This PR deprecates `Schedule.instructions` for `Schedule.timed_Instructions(flatten=True)`, which will enable us to move towards a Schedule tree with `Call` instructions.

### CODE REVIEW

File: qiskit/assembler/assemble_schedules.py [hunk 1]
```diff
@@ -93,6 +93,9 @@ def _assemble_experiments(
                                     converters.InstructionToQobjConverter)
     instruction_converter = instruction_converter(qobj.PulseQobjInstruction,
                                                   **run_config.to_dict())
+
+    schedules = [sched if isinstance(sched, pulse.Schedule) else pulse.Schedule(sched)
+                 for sched in schedules]
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:98]: Isn't this because of https://github.com/Qiskit/qiskit-terra/issues/4523? Could we put this in a separate PR and merge it quickly in this release?

File: qiskit/pulse/interfaces.py [hunk 2]
```diff
@@ -73,14 +73,13 @@ def ch_stop_time(self, *channels: List[Channel]) -> int:
...
+    def _children(self) -> List[Union[int, 'ScheduleComponent']]:
         """Child nodes of this schedule component. """
         pass
 
-    @property
     @abstractmethod
-    def instructions(self) -> Tuple[Tuple[int, 'Instructions']]:
-        """Return iterable for all `Instruction`s in `Schedule` tree."""
+    def timed_instructions(self) -> List[Tuple[int, 'Instructions']]:
+        """Return tuple of all `Instruction`s with their time in the `Schedule` tree."""
```
Comment (lcapelluto) [inline qiskit/pulse/interfaces.py:82]: Is it a tuple or a list? Why switch back?

**Comment by taalexander** [comment]
> I'm a little confused about what this is accomplishing. `instructions` is a good and useful attribute.
> 
> Why accept `flatten` as an argument? The behavior is separate, isn't it? Rather than `sched.timed_instructions(flatten=True)`, the user should write `sched.flatten().timed_instructions` if they want it "flattened". (Although I had thought that `instructions` were always returned flat)

The goal is to not return the time of every instruction. This will allow us to move to a representation that does not have absolute timing and Schedule instructions will be able to return the top-level instructions. This will enable code re-usue by only returning the top-level elements in the schedule tree.

I added flatten to maintain the current behavior. I don't want to return the inlined schedule tree but want to allow referencing of other schedules from within a schedule, these references should be made through a `Call` instruction which will be enabled once we go through this deprecation period.

File: qiskit/assembler/assemble_schedules.py (hunk 1 above)
Comment (taalexander) [reply qiskit/assembler/assemble_schedules.py:98]: Oh you're right. Didn't realize this was a separate bug.

Comment (taalexander) [reply qiskit/assembler/assemble_schedules.py:98]: Added PR #4870

**Comment by taalexander** [comment]
Closing in favour of addressing this in the future IR.
